service/src/main/java/eu/europa/ec/fisheries/uvms/activity/service/bean/ActivityServiceBean.java

```diff
@@ -398,14 +405,53 @@ public int getPreviousFishingActivity(int fishingActivityId) {
         log.info("Retrieve fishing activity from db: {}", fishingActivityId);
         FishingActivityEntity activityEntity = fishingActivityDao.getFishingActivityById(fishingActivityId, null);
         log.info("activityEntity received from db Id: {} typeCode: {} Date: {}", activityEntity.getId(), activityEntity.getTypeCode(), DateUtils.parseUTCDateToString(activityEntity.getCalculatedStartTime()));
-        return fishingActivityDao.getPreviousFishingActivityId(activityEntity);
+
+        FishingTripEntity fishingTrip = activityEntity.getFishingTrip();
+        Set<FishingActivityEntity> fishingActivities = fishingTrip.getFishingActivities();
+
+        List<FishingActivityEntity> fishingActivityEntityList = new ArrayList<>(fishingActivities);
+        fishingActivityEntityList.sort(Comparator.comparing(FishingActivityEntity::getOccurence).reversed());
+
+        int index = -1;
+        for (int i = 0; i < fishingActivityEntityList.size(); i++) {
```

Comment:
And if you do "for ([...] i < fishingActivityEntityList.size() - 1; [...]" then you can do you return statement inside the if clause in this for loop. I'm just full of micro optimizations that probably will not make any difference today!